docs/style.md

```diff
@@ -269,3 +281,191 @@ When writing event handlers, we prefer using the exact name, `onClick` to a
 shorthand. If that name is already being used in a given scope, which often
 happens if the component supports a prop `onClick`, then we prefer to specify
 the function as `handleOnClick`.
+
+## Sass
+
+### Guidelines
+
+#### Author component styles using mixins
+
+When authoring the styles for a component, it's important that we use
+[`mixins`](https://sass-lang.com/documentation/at-rules/mixin) to allow
+developers to control when the CSS for a specific component gets emitted. For
+example:
+
+```scss
+// src/components/accordion/_accordion.scss
+
+/// Accordion
+/// @access private
+/// @group accordion
+@mixin accordion {
+  .#{$prefix}--accordion {
+    // ...
+  }
+}
+```
+
+Authoring component styles under a mixin allows the design system to:
+
+- Control when the CSS for accordion gets emitted
+- Allows us to author experimental or future styles in a separate mixin and
+  toggle its inclusion through feature flags
+- Could allow developers consuming the design system to control when styles get
+  emitted
+
+#### Use design tokens where appropriate
+
+We have a number of Sass variables available in the project to be used as design
+tokens when building out a component. Almost always you will want to leverage
+these instead of hard coding values for colors, type, or even spacing. You can
+visit the following SassDoc links to view all of the design tokens relevant to
+this project:
+
+- [Color](../packages/theme/docs/sass.md)
+- [Layout](../packages/layout/docs/sass.md)
+- [Motion](../packages/motion/docs/sass.md)
+- [Type](../packages/type/docs/sass.md)
+
+#### Avoid nesting selectors
+
+Nesting selectors is often a convenient and fast way to author styles in Sass.
+Unfortunately, they also add a performance and maintenance burden for the design
+system. The performance burden is due to the generated nature of selectors which
+can lead to unexpected CSS bundle bloat. The maintenance burden manifests itself
+in a way that makes it harder to find specific selectors while working on the
+codebase. For example, if we are looking for the selector `.component:focus` in
+the following file:
+
+```scss
+// Early on in the file
+.component {
+  // ...
+}
+
+// ...
+
+// Later on in the file
+.component {
+  &:focus {
```

Comment:
CC @tw15egan

Reply:
I really don't have a problem with not nesting selectors, my main gripe is just having inconsistent styling throughout the repo. Unnesting all selectors should be part of a major refactor if searchability is the main concern.

Reply:
@tw15egan 100% agreed!